demisto_sdk/commands/generate_outputs/generate_descriptions/generate_descriptions.py

```diff
@@ -0,0 +1,289 @@
+    # Learn from mistakes
+    if final_output != output:
+        y = input("Should we append to prompt (y/n)? ")
+        if y == 'y':
+            print(
+                f"(Correcting prompt `{final_output}` instead of `{output}`)")
+            correct_prompt(command_ctx, final_output)
+    return final_output
+
+
+def generate_ai_descriptions(
+        input_path: str,
+        output_path: str = "out.yml",
+        interactive: bool = True,
+        verbose: bool = False,
+        insecure: bool = False,
+):
+    """ Generate integration command contexts.
+
+    Args:
+        input_path: path to the yaml integration input path
+        output_path: path to the yaml integration output path
+        interactive: interactivity (correct ai result mistakes)
+        verbose: verbose (debug mode)
+        insecure: insecure https (debug mode)
+    """
+    print()
+    print("⚠" * 100)
+    print('''███████ ██   ██ ██████  ███████ ██████  ██ ███    ███ ███████ ███    ██ ████████  █████  ██
+██       ██ ██  ██   ██ ██      ██   ██ ██ ████  ████ ██      ████   ██    ██    ██   ██ ██
+█████     ███   ██████  █████   ██████  ██ ██ ████ ██ █████   ██ ██  ██    ██    ███████ ██
+██       ██ ██  ██      ██      ██   ██ ██ ██  ██  ██ ██      ██  ██ ██    ██    ██   ██ ██
+███████ ██   ██ ██      ███████ ██   ██ ██ ██      ██ ███████ ██   ████    ██    ██   ██ ███████
+''')
+    print("this feature is experimental, use with caution.")
+    print("⚠" * 100)
+    print()
+
+    try:
+        similar_paths: Dict[str, str] = {}
+        yml_data = get_yaml(input_path)
+
+        # Fix sometimes the yml doesn't contain the full structure
+        #  (json-to-outputs)
+        if 'script' not in yml_data:
+            yml_data = {'script': {'commands': [yml_data]}}
+
+        commands = yml_data.get("script", {})
+        commands = commands.get('commands', [])
+
+        # Iterate over every command
+        for c_index, command in enumerate(commands):
+            command_name = command.get('name')
+
+            if interactive or verbose:
+                print(f'Command: {command_name}')
+
+            outputs = command.get('outputs')
+            if not outputs:
+                if interactive or verbose:
+                    print("-- Skipping because no outputs for command")
+                continue
+
+            # Iterate over every output per command
+            for o_index, o in enumerate(outputs):
+                command_ctx = o.get('contextPath')
+
+                # Match past paths automatically.
+                if command_ctx in similar_paths:
+                    print(
+                        f"--Already added description for exact path: {command_ctx}--")
+                    final_output = similar_paths.get(command_ctx)
+                    yml_data['script']['commands'][c_index]['outputs'][o_index][
+                        'description'] = final_output
+                    write_yml(output_path, yml_data)
```

Comment:
I wonder if we want to write to file each time instead of writing the whole YML after the whole iteration. WDYT?

Reply:
Sometimes you need to input 100s of descriptions, so I like it to be backed up all the time incrementally. For example if something goes wrong, I still want the old descriptions (that we used the API for).

Reply:
Whats your opinion here? @yuvalbenshalom . Both approaches has pros and cons

Reply:
Why does it matter though? it's not like an optimization will help (it's still fast enough not to notice even big files). As you work one at a time and wait 1-2 seconds for the GPT response in between (plenty of time for the disk IO)